### PR TITLE
test: make p2p network tests work on Intel macOS hosts

### PR DESCRIPTION
**Pull request by rudolfs**
So we can get rid of the dev vm on gcp.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    apt-get -y upgrade
+    apt-get -y --no-install-recommends install \
+      build-essential \
+      ca-certificates \
+      cmake \
+      curl \
+      git \
+      jq \
+      procps; \
+    rm -rf /var/lib/apt/lists/*;
```
Comment (geigerzaehler) [inline Vagrantfile:34]: I don’t think this is necessary. It saves a bit of space but makes subsequent apt commands slower.

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    curl -sfLS https://deb.nodesource.com/setup_14.x | bash -; \
+    apt-get -y update; \
+    apt-get -y --no-install-recommends install nodejs; \
+    apt-get autoremove; \
+    rm -rf /var/lib/apt/lists/*; \
```
Comment (geigerzaehler) [inline Vagrantfile:40]: See comment above

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    echo "${RUSTUP_SHA256} *rustup-init" | sha256sum -c -; \
+    chmod +x rustup-init; \
+    ./rustup-init -y --no-modify-path --profile minimal --default-toolchain $RUST_VERSION; \
+    rm rustup-init; \
+    chmod -R a+w $RUSTUP_HOME $CARGO_HOME; \
+    rustup --version; \
+    cargo --version; \
+    rustc --version; \
+    rm -rf /usr/local/cargo/registry; \
+    rm -rf /usr/local/cargo/.package-cache;
```
Comment (geigerzaehler) [inline Vagrantfile:60]: I think we can simplify this a bit
```bash
curl https://sh.rustup.rs -sSfL \
 | sudo -u vagrant sh -s -- -y --profile minimal
```

No need to do all the docker image specific stuff. We can also remove specifing the rust version explicitly. `rustup` will take it from the file anyways and this way we don’t need to remember to update the version here if we bump the toolchain.

File: Vagrantfile [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    echo "PATH=/usr/local/cargo/bin:$PATH" >> /home/vagrant/.bashrc
+    echo "export CARGO_HOME=/home/vagrant/cargo" >> /home/vagrant/.bashrc
+    echo "export CARGO_TARGET_DIR=/home/vagrant/cargo" >> /home/vagrant/.bashrc
```
Comment (geigerzaehler) [inline Vagrantfile:64]: I don’t know if its a good idea to have them point to the same directory. Also not sure we need to set `CARGO_HOME`. The default value should be sufficient.

File: Vagrantfile (hunk 1 above)
Comment (rudolfs) [reply Vagrantfile]: 65e982ac431e796be804a71bc79cd2b51fe5e6fc

File: Vagrantfile (hunk 2 above)
Comment (rudolfs) [reply Vagrantfile]: 65e982ac431e796be804a71bc79cd2b51fe5e6fc

File: Vagrantfile (hunk 3 above)
Comment (rudolfs) [reply Vagrantfile]: 351b3371f953c28460d53474b152be63bce9435f

File: Vagrantfile (hunk 4 above)
Comment (rudolfs) [reply Vagrantfile]: 351b3371f953c28460d53474b152be63bce9435f

**Review by geigerzaehler — APPROVED** [review]

**Pull request merged**